File: Client/src/app/models/property.ts
export interface PropertyImageDto {
  id: number;
  imageUrl: string;  // Changed from 'url' to match backend
  isPrimary: boolean;
  category?: string;  // Added to match backend
}

export interface Amenity {
  id: number;
  name: string;
  category: string;
  iconUrl: string;
}

export interface CancellationPolicy {
  id: number;
  name: string;
  description: string;
  refundPercentage: number;
}

export interface PropertyDto {
  id: number;
  hostId?: number;
  title: string;
  cancellationPolicyId: number;
  cancellationPolicy?: CancellationPolicy;
  description: string;
  propertyType: string;
  address: string;
  categoryId: number;
  city: string;
  country?: string;  // Made optional
  latitude?: number;
  longitude?: number;
  pricePerNight?: number;
  cleaningFee?: number;
  serviceFee?: number;
  minNights?: number;
  maxNights?: number;
  bedrooms?: number;
  bathrooms?: number;
  maxGuests?: number;
  status?: string;
  createdAt?: Date;
  updatedAt?: Date;
  isAvailable?: boolean;
  images: PropertyImageDto[];
  amenities?: Amenity[];  // Added amenities array
  
  // Rating and favorite fields
  averageRating?: number;
  reviewCount?: number;
  favoriteCount?: number;
  isGuestFavorite?: boolean;
  
  // UI-only fields
  viewCount?: number;
  dates?: string;
  currentImageIndex?: number; // Optional property for carousel functionality
  instantBook?: boolean; // Added instantBook property
}

export interface PropertyCategory {
  categoryId: number;
  name: string;
  description?: string;
  iconUrl: string;
}

File: API/Models/Property.cs
namespace API.Models
{
    public enum PropertyStatus
    {
        Active,
        Pending,
        Suspended,
    }
    public class Property
    {
        public int Id { get; set; }
        public int HostId { get; set; }
        public int CategoryId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string PropertyType { get; set; }
        public string Country { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string PostalCode { get; set; }
        public decimal Latitude { get; set; }
        public decimal Longitude { get; set; }
        public string Currency { get; set; } = "USD";
        public decimal PricePerNight { get; set; }
        public decimal CleaningFee { get; set; }
        public decimal ServiceFee { get; set; }
        public int MinNights { get; set; } = 1;
        public int MaxNights { get; set; }
        public int Bedrooms { get; set; }
        public int Bathrooms { get; set; }
        public int MaxGuests { get; set; }
        public TimeSpan? CheckInTime { get; set; }
        public TimeSpan? CheckOutTime { get; set; }
        public bool InstantBook { get; set; } = false;
        public string Status { get; set; } = PropertyStatus.Pending.ToString();
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } 
        public int CancellationPolicyId { get; set; }

        // Navigation Properties
        public virtual Host Host { get; set; } = null!;
        public virtual CancellationPolicy CancellationPolicy { get; set; }
        public virtual PropertyCategory Category { get; set; }
        public virtual ICollection<PropertyImage> PropertyImages { get; set; } = new List<PropertyImage>();
        public virtual ICollection<Amenity> Amenities { get; set; } = new List<Amenity>();
        public virtual ICollection<Booking> Bookings { get; set; } = new List<Booking>();
        public virtual ICollection<Favourite> Favourites { get; set; } = new List<Favourite>();
        public virtual ICollection<PropertyAvailability> Availabilities { get; set; } = new List<PropertyAvailability>();
        public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();
        public virtual ICollection<Conversation> Conversations { get; set; } = new List<Conversation>();


    }
}
